master/internal/db/static_query_map.go

```diff
@@ -0,0 +1,26 @@
+package db
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/determined-ai/determined/master/pkg/etc"
+)
+
+type staticQueryMap struct {
+	queries map[string]string
+	sync.RWMutex
+}
+
+func (q *staticQueryMap) getOrLoad(queryName string) string {
+	q.RLock()
+	query, ok := q.queries[queryName]
```

Comment:
FWIW, as a practical matter, I'm not sure this optimization is warranted -- i.e.,

```
q.Lock()
defer q.Unlock()
query, ok := q.queries[queryName]
if !ok {
    query = string(etc.MustStaticFile(...))
    q.queries[queryName] = query
}
return query
```

would probably be fine in practice.

Reply:
I thought that for this specific case with 10-20 writes and unbounded reads, this would be better. But I did benchmark them for fun and they were very close. I can change it since it's so much simpler.

Reply:
Went ahead and changed this, since it's impact is minimal in comparison to the db query associated with each call.